### PR TITLE
Update InlineLink styling

### LINKED ISSUE #286: Update InlineLink appearance
**Is your feature request related to a problem? Please describe.**
UX have changed the designs for inline links.

~Blocked on #277.~ Done.

**Describe the solution you'd like**
Following this, the InlineLink component should have the following stylings, whether included in a Text component or in a Caption component:

**Default**
Text colour: Shadow (`#3F3F42`)
Underline: 1px Postbox (`#B80000`) (text-decoration, not a CSS border)

**Hover**
Text colour: Postbox (`#B80000`)
Underline: 2px Postbox (`#B80000`) (text-decoration, not a CSS border) 

**Visited**
Text colour: Cloud-dark (`#757575`)
Underline: 1px Cloud-dark (`#757575`) (text-decoration, not a CSS border) 

**Describe alternatives you've considered**

**Testing notes**
Test re colour blindness, WCAG colour ratio. Test that the different states display as expected.

Dev insight: Please check appearance when contained in both a Text component & a Caption component. Update snapshots and view in Storybook. 

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by bcmn**
Resolves #286 

**Overall change:** Changes the InlineLink appearance to match the updated Simorgh designs.

(Screenshots npm linked into psammead-caption to help display it better)

Default
screenshot 2019-02-26 at 09 07 26

Hover/Focus
screenshot 2019-02-26 at 09 07 37

Visited
screenshot 2019-02-26 at 09 07 46

**Code changes:**

- Updates InlineLink colours, bumping psammead-styles dependency.
- I removed the padding from the component, as I believe this was part of the previous design, with the background?
- Currently doesn't alter underline implementation -- I think we wanted to avoid using `border-bottom` due to issues with Reith overlapping, but UX have specified specific pixel stylings depending on hover/visited state. 🤔 This overlapping seemed to still occur slightly when I experimented with `box-shadow` as an alternative implementation.

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests ~added~ updated for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-inline-link/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 | Version | Description |
 |---------|-------------|
+| 0.3.0   |  |
```
Comment (sareh) [inline packages/components/psammead-inline-link/CHANGELOG.md]: Could you please update the changelog here?

File: packages/components/psammead-inline-link/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bbc/psammead-inline-link",
-  "version": "0.2.0",
+  "version": "0.3.0",
```
Comment (sareh) [inline packages/components/psammead-inline-link/package.json:3]: Please update the `package-lock.json` too.

**Review by sareh — CHANGES_REQUESTED** [review]

File: packages/components/psammead-inline-link/package.json (hunk 2 above)
Comment (sareh) [reply packages/components/psammead-inline-link/package.json:3]: The comment under the `roadmap` can likely be removed, too: https://github.com/bbc/psammead/blob/latest/packages/components/psammead-inline-link/README.md#roadmap

**Review by sareh — APPROVED** [review]
👍 Looks great! Thanks for making the changes.

**Review by pjlee11 — APPROVED** [review]
LGTM 👍

**Comment by jamesbrumpton** [comment]
👍

**Pull request merged**